wrapper/src/main/java/software/amazon/jdbc/PluginServiceImpl.java

```diff
@@ -65,32 +68,72 @@ public class PluginServiceImpl implements PluginService, CanReleaseResources,
   private boolean isInTransaction;
   private boolean explicitReadOnly;
   private final ExceptionManager exceptionManager;
+  protected final @Nullable ExceptionHandler exceptionHandler;
   protected final DialectProvider dialectProvider;
   protected Dialect dialect;
+  protected TargetDriverDialect targetDriverDialect;
+  protected @Nullable final ConfigurationProfile configurationProfile;
 
   public PluginServiceImpl(
       @NonNull final ConnectionPluginManager pluginManager,
       @NonNull final Properties props,
       @NonNull final String originalUrl,
-      final String targetDriverProtocol) throws SQLException {
-    this(pluginManager, new ExceptionManager(), props, originalUrl, targetDriverProtocol,
+      @NonNull final String targetDriverProtocol,
+      @NonNull final TargetDriverDialect targetDriverDialect)
+      throws SQLException {
+
+    this(pluginManager,
+        new ExceptionManager(),
+        props,
+        originalUrl,
+        targetDriverProtocol,
+        null,
+        targetDriverDialect,
         null);
   }
 
+  public PluginServiceImpl(
+      @NonNull final ConnectionPluginManager pluginManager,
+      @NonNull final Properties props,
+      @NonNull final String originalUrl,
+      @NonNull final String targetDriverProtocol,
+      @NonNull final TargetDriverDialect targetDriverDialect,
+      @Nullable final ConfigurationProfile configurationProfile) throws SQLException {
+    this(pluginManager,
+        new ExceptionManager(),
+        props,
+        originalUrl,
+        targetDriverProtocol,
+        null,
+        targetDriverDialect,
+        configurationProfile);
+  }
+
   public PluginServiceImpl(
       @NonNull final ConnectionPluginManager pluginManager,
       @NonNull final ExceptionManager exceptionManager,
       @NonNull final Properties props,
       @NonNull final String originalUrl,
-      final String targetDriverProtocol,
-      @Nullable final DialectProvider dialectProvider) throws SQLException {
+      @NonNull final String targetDriverProtocol,
+      @Nullable final DialectProvider dialectProvider,
+      @NonNull final TargetDriverDialect targetDriverDialect,
+      @Nullable final ConfigurationProfile configurationProfile) throws SQLException {
     this.pluginManager = pluginManager;
     this.props = props;
     this.originalUrl = originalUrl;
     this.driverProtocol = targetDriverProtocol;
+    this.configurationProfile = configurationProfile;
     this.exceptionManager = exceptionManager;
     this.dialectProvider = dialectProvider != null ? dialectProvider : new DialectManager(this);
-    this.dialect = this.dialectProvider.getDialect(this.driverProtocol, this.originalUrl, this.props);
+    this.targetDriverDialect = targetDriverDialect;
+
+    this.exceptionHandler = this.configurationProfile != null && this.configurationProfile.getExceptionHandler() != null
```

Comment:
if we're returning null when getExceptionHandler is null,
then we could just check the profile

this.configurationProfile != null ? this.configurationProfile.getExceptionHandler() : null